### PR TITLE
Fix assert

### PR DESCRIPTION
**Pull request by coderhaoxin**

### CODE REVIEW

File: lib/collection/query/update.js [hunk 1]
```diff
@@ -76,8 +76,10 @@ Query.prototype.promiseUpdate = function () {
   options.multi = true
   return new Promise(function (resolve, reject) {
     self.collection.update(self.criteria, self.document, options, function (err, results) {
-      if (err) reject(err)
-      else resolve(results)
+      if (err) return reject(err)
+      var result = results.result || results
+      if (result.nModified != null) return resolve(result.nModified)
+      resolve(result)
     })
```
Comment (coderhaoxin) [inline lib/collection/query/update.js]: curious about why the results is

```js
{ 
  result: { ok: 1, ... },
  connection: {...},
  ...
```

**Comment by jonathanong** [comment]
whoops. yeah some v2 changes.

File: lib/collection/query/update.js (hunk 1 above)
Comment (jonathanong) [inline lib/collection/query/update.js]: only issue i can think of is when a document actually has `.result` as a property. are there more properties we can check?

Comment (coderhaoxin) [inline lib/collection/query/update.js]: .update() not return the document.
the `results` from node-mongodb-native@2 is something like

```js
{
  result: {
    ok: 1,
    nModified: 1,
    n: 1
  },
  connection: {...}
}
```

Comment (jonathanong) [inline lib/collection/query/update.js]: actually this is only when updating documents... yeah it's fine.

**Pull request merged**